README.md

```diff
@@ -228,6 +228,48 @@ will spawn a number of threads in order to optimize performance for
 iRODS server versions 4.2.9+ and file sizes larger than a default
 threshold value of 32 Megabytes.
 
+Progress bars
+-------------
+
+The PRC now has support for progress bars which function on the basis of
+an "update" callback function.  In the case of a tqdm progress bar, you can always just
+pass the update method of the progress bar instance directly to the data_object
+`put` or `get` method:
+
+```
+   pbar = tqdm.tqdm(total = data_obj.size)
+   session.data_objects.get(file_name, data_obj.path, updatables = pbar.update)
+```
+
+The updatables parameter can be a list or tuple of update-enabling objects and/or bound methods.
+
+Alternatively, the tqdm progress bar object itself can be passed in, if an adapting
+function such as the following is first registered:
+
+```
+    def adapt_tqdm(pbar, l = threading.Lock()):
+        def _update(n):
+            with l:
+                pbar.update(n)
+        return _update
+    irods.manager.data_objects_manager.register_update_type( adapt_tqdm )
+    session.data_objects.put( file, logical_path, updatables = [tqdm_1,tqdm_2] ) # update two tqdm's simultaneously
+```
+
+Other progress bars may be included in an updatables parameter, but may require more extensive adaptation.
+For example, the ProgressBar object (from the progressbar module) also has an update method, but this one
```

Comment:
That looks like an update to the progressbar that I was using previously.  Fortunately, it looks as if progressbar2 objects actually can be pointed to by weak references, so they must have resolved that since the predecessor version of `progressbar`.

Reply:
linked.